File: Movie Details/MovieDetails/app/src/main/java/com/example/moviedetails/MainActivity.java
package com.example.moviedetails;

import android.os.Bundle;
import android.widget.Button;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

public class MainActivity extends AppCompatActivity {

    Button btnBasic, btnAdditional;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        btnBasic = findViewById(R.id.btnBasic);
        btnAdditional = findViewById(R.id.btnAdditional);

        btnBasic.setOnClickListener(v -> loadFragment(new MovieBasicFragment()));

        btnAdditional.setOnClickListener(v -> loadFragment(new MovieAdditionalFragment()));
    }

    private void loadFragment(Fragment fragment) {
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.fragmentContainer, fragment)
                .commit();
    }
}

File: StudentFragmentApp/app/src/main/java/com/example/studentfragmentapp/MainActivity.kt
package com.example.studentfragmentapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button

class MainActivity : AppCompatActivity() {

    lateinit var btnBasic: Button
    lateinit var btnMark: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnBasic = findViewById(R.id.btnBasic)
        btnMark = findViewById(R.id.btnMark)

        btnBasic.setOnClickListener {

            val fragment = StudentBasicFragment()
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, fragment)
                .commit()
        }

        btnMark.setOnClickListener {

            val fragment = StudentMarkFragment()
            supportFragmentManager.beginTransaction()
                .replace(R.id.fragment_container, fragment)
                .commit()
        }
    }
}